packages/wonder-blocks-popover/src/components/__tests__/popover.test.tsx

```diff
@@ -602,19 +606,67 @@ describe("Popover", () => {
             );
 
             // Act
-            // Open the popover
-            const openButton = await screen.findByRole("button", {
-                name: "Open default popover",
-            });
-
-            await userEvent.click(openButton);
-            const popover = await screen.findByRole("dialog");
+            await userEvent.click(
+                await screen.findByRole("button", {
+                    name: "Open default popover",
+                }),
+            );
 
             // Assert
+            expect(
+                await screen.findByRole("dialog", {
+                    name: "Popover Aria Label",
+                }),
+            ).toBeInTheDocument();
 
-            expect(popover).toHaveAttribute("aria-label", "Popover Aria Label");
-            expect(popover).not.toHaveAttribute("aria-labelledby");
-            expect(popover).not.toHaveAttribute("aria-describedby");
+            expect(
+                await screen.findByRole("dialog", {
+                    description: "This is a popover description",
```

Comment:
**suggestion:** With this refactor in place, I think it makes sense to create a separate test to assert the `description` part of this test.

For some context, we try to follow the AAA approach for testing: https://khanacademy.atlassian.net/wiki/spaces/ENG/pages/98402353/JavaScript+Testing+Best+Practices#Assert-One-Requirement-Per-Test
https://khanacademy.atlassian.net/wiki/spaces/ENG/pages/98402353/JavaScript+Testing+Best+Practices#Assert-One-Requirement-Per-Test

One of the main benefits we have found by using this approach is that it's easier to isolate a failure when it happens and also easier to apply the fix when needed.

Reply:
Thanks Juan, I split it out into three tests that each have 1 assertion